### DISCUSSION TITLE
Version 2 of favorites, routes, tracks and profiles

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#585] [Q&A] [answered]
I am thinking of ways to make the storage of favorites, routes, tracks more solid.

Question: what number of favorites / routes / tracks do you usually store?

If someone has a large amount of this data and wants to email it to me,
it would be useful for testing. :slightly_smiling_face:

**Comment by umpfmpf** [comment]
To which email address?

**Reply by devemux86** [reply]
@umpfmpf 

The contact email is listed on [Google Play](https://play.google.com/store/apps/details?id=gr.talent.cruiser) in the "App support" section.

**Reply by umpfmpf** [reply]
done

**Reply by devemux86** [reply]
@umpfmpf 

Are you sure? I do not see anything.

**Reply by umpfmpf** [reply]
Yes. But it is 12MB

**Reply by devemux86** [reply]
@umpfmpf 

It was in spam. :slightly_smiling_face: 

These are not Cruiser folders, right? They are collections of favorites and tracks.

**Reply by umpfmpf** [reply]
Cant you use it?

**Reply by devemux86** [reply]
@umpfmpf  thanks, it's fine!

(just wanted to clarify what is the content)

**Comment by DolomitesTom** [comment]
done already

**Comment by airald** [comment]
Sent my POI collection. About 300 places in 30 groups.

**Comment by devemux86** [comment]
Next (Beta) update will advance the storage of favorites, routes, tracks to v2 with a simpler implementation.
When next update first starts, it will automatically convert the storage to v2 (folders will have the "2" suffix).

---

_**As always with all Beta versions, before updating the app, please BACKUP your favorites / routes / tracks and profiles!**_

**Reply by devemux86** [reply]
Profiles will also advance to v2. Some of the benefits are:
- All profiles are considered user profiles
- We can rename and delete all profiles
- We can keep only the necessary profiles
- We can create new profiles with any type of travel

**Reply by devemux86** [reply]
The tracks will also be stored in a new format, json instead of gpx, just like the routes.

The speed improvement is huge:
- A large collection of 25MB gpx tracks can now take 7" to process
- With v2 it will only take less than 1"

**Comment by devemux86** [comment]
Cruiser 5 (Beta) advances the storage of favorites, routes, tracks, profiles to v2 with a new implementation.
When next update first starts, it will automatically convert the storage to v2 (folders will have the "2" suffix).

- https://github.com/devemux86/cruiser/discussions/610

---

_**Before updating the app, please BACKUP your favorites / routes / tracks and profiles!**_

**Comment by devemux86** [comment, marked as answer]
- Version 2 of favorites, routes, tracks and profiles is available in https://github.com/devemux86/cruiser/discussions/641
